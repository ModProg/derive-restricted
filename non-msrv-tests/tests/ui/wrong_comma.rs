#[derive_where::derive_where(Clone,; T)]
struct Test1<T>(T);

#[derive_where::derive_where(Clone,,)]
struct Test2<T>(T);

#[derive_where::derive_where(Clone; T;)]
struct Test3<T>(T);

#[derive_where::derive_where(Clone; T,,)]
struct Test4<T>(T);

fn main() {}
