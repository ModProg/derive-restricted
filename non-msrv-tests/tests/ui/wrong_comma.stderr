error: expected one of Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Zeroize
 --> tests/ui/wrong_comma.rs:1:36
  |
1 | #[derive_where::derive_where(Clone,; T)]
  |                                    ^

error: expected one of Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Zeroize
 --> tests/ui/wrong_comma.rs:4:36
  |
4 | #[derive_where::derive_where(Clone,,)]
  |                                    ^

error: expected `,`
 --> tests/ui/wrong_comma.rs:7:38
  |
7 | #[derive_where::derive_where(Clone; T;)]
  |                                      ^

error: expected type to bind to, expected one of: `for`, parentheses, `fn`, `unsafe`, `extern`, identifier, `::`, `<`, square brackets, `*`, `&`, `!`, `impl`, `_`, lifetime
  --> tests/ui/wrong_comma.rs:10:39
   |
10 | #[derive_where::derive_where(Clone; T,,)]
   |                                       ^
